From 9b23f18d94d7a1ddb01bbfd1e49eefbc188794d7 Mon Sep 17 00:00:00 2001
From: Baji Patthan <bpatth@codeaurora.org>
Date: Tue, 5 May 2015 19:33:43 -0700
Subject: [PATCH 08/22] ARM:dts:msm: Add support for SBC8016 P2

1) Fix switch between the USB GPIOs and LEDs GPIOs.
2) Add support for USB on the fly switch between device and host mode.

Change-Id: I391afa85ab8a43b23fed016f304802d17ba58d95
Signed-off-by: Baji Patthan <bpatth@codeaurora.org>
---
 arch/arm/boot/dts/qcom/apq8016-sbc.dtsi | 57 +++++++++++++++++----------------
 1 file changed, 30 insertions(+), 27 deletions(-)

diff --git a/arch/arm/boot/dts/qcom/apq8016-sbc.dtsi b/arch/arm/boot/dts/qcom/apq8016-sbc.dtsi
index f97356f..545b9dd 100644
--- a/arch/arm/boot/dts/qcom/apq8016-sbc.dtsi
+++ b/arch/arm/boot/dts/qcom/apq8016-sbc.dtsi
@@ -271,6 +271,8 @@
 &usb_otg {
 	qcom,hsusb-otg-mode = <3>;
 	qcom,usbid-gpio = <&msm_gpio 121 0>;
+	qcom,hub-reset-gpio = <&pm8916_gpios 3 0>;
+	qcom,sw-sel-gpio = <&pm8916_gpios 4 0>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&usbid_default>;
 	vbus_otg-supply = <&vph_pwr_vreg>;
@@ -521,27 +523,7 @@
 };
 
 &pm8916_gpios {
-	/* USB_HUB_RESET_N*/
 	gpio@c000 { /* GPIO 1 */
-		qcom,mode = <0>;	/* DIGITAL IN */
-		qcom,pull = <2>;	/* PULL UP */
-		qcom,vin-sel = <3>;	/* VPH */
-		qcom,src-sel = <0>;	/* CONSTANT */
-		qcom,master-en = <1>; 	/* ENABLE GPIO */
-		status = "okay";
-	};
-
-	/* USB_SW_SEL*/
-	gpio@c100 { /* GPIO 2 */
-		qcom,mode = <0>;	/* DIGITAL IN */
-		qcom,pull = <5>;	/* No PULL */
-		qcom,vin-sel = <0>;	/* VPH */
-		qcom,src-sel = <0>;	/* CONSTANT */
-		qcom,master-en = <1>; 	/* ENABLE GPIO */
-		status = "okay";
-	};
-
-	gpio@c200 { /* GPIO 3 */
 		qcom,mode = <1>;		/* Digital output */
 		qcom,pull = <4>;		/* Pull down */
 		qcom,vin-sel = <2>;		/* VIN 2*/
@@ -550,7 +532,7 @@
 		status = "okay";
 	};
 
-	gpio@c300 { /* GPIO 4 */
+	gpio@c100 { /* GPIO 2 */
 		qcom,mode = <1>;		/* Digital output */
 		qcom,pull = <4>;		/* Pull down */
 		qcom,vin-sel = <2>;		/* VIN 2*/
@@ -558,15 +540,35 @@
 		qcom,master-en = <1>;	/* Enable GPIO */
 		status = "okay";
 	};
+
+	/* USB_HUB_RESET_N*/
+	gpio@c200 { /* GPIO 3*/
+		qcom,mode = <1>;	/* Digital output */
+		qcom,pull = <5>;	/* No PULL */
+		qcom,vin-sel = <0>;	/* VPH */
+		qcom,src-sel = <0>;	/* CONSTANT */
+		qcom,master-en = <1>; 	/* ENABLE GPIO */
+		status = "okay";
+	};
+
+	/* USB_SW_SEL*/
+	gpio@c300 { /* GPIO 4 */
+		qcom,mode = <1>;	/* Digital output */
+		qcom,pull = <5>;	/* No PULL */
+		qcom,vin-sel = <0>;	/* VPH */
+		qcom,src-sel = <0>;	/* CONSTANT */
+		qcom,master-en = <1>; 	/* ENABLE GPIO */
+		status = "okay";
+	};
 };
 
 &spmi_bus {
 	qcom,pm8916@0 {
-		qcom,leds@c200  {
+		qcom,leds@c000  {
 			compatible = "qcom,leds-qpnp";
-			reg = <0xc200 0x100>;
+			reg = <0xc000 0x100>;
 			status = "okay";
-			qcom,led_gpio_3 {
+			qcom,led_gpio_1 {
 				label = "gpio";
 				linux,name = "led3";
 				qcom,max-current = <40>;
@@ -578,11 +580,11 @@
 				qcom,vin-ctrl = <0x02>;
 			};
 		};
-		qcom,leds@c300  {
+		qcom,leds@c100  {
 			compatible = "qcom,leds-qpnp";
-			reg = <0xc300 0x100>;
+			reg = <0xc100 0x100>;
 			status = "okay";
-			qcom,led_gpio_4 {
+			qcom,led_gpio_2 {
 				label = "gpio";
 				linux,name = "boot";
 				qcom,max-current = <40>;
@@ -636,6 +638,7 @@
 	qcom,init-voltage = <1300000>;
 };
 
+
 &spmi_bus {
 	qcom,pm8916@0 {
 		qcom,leds@a100 { /* WLAN LED */
-- 
1.8.2.1

